zerver/tests/test_archive.py

```diff
@@ -80,3 +80,39 @@ def test_get_web_public_streams(self) -> None:
         self.assert_length(public_subs, 1)
         self.assert_length(public_unsubs, 0)
         self.assert_length(public_neversubs, 0)
+
+class WebPublicTopicHistoryTest(ZulipTestCase):
+    def test_non_existant_stream_id(self) -> None:
+        result = self.client_get("/archive/100000000/topics")
+        self.assert_json_success(result)
+        history = result.json()['topics']
+        self.assertEqual(history, [])
+
+    def test_non_web_public_stream(self) -> None:
+        test_stream = self.make_stream('Test Public Archives')
+        result = self.client_get(
+            "/archive/" + str(test_stream.id) + "/topics"
+        )
+        self.assert_json_success(result)
+        history = result.json()['topics']
+        self.assertEqual(history, [])
```

Comment:
This test should send a stream message, otherwise the history would be empty even if the stream wasn't tagged as web-public.